dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteMemoryStore.cs

```diff
@@ -90,17 +90,9 @@ public async IAsyncEnumerable<string> UpsertBatchAsync(string collectionName, IE
     public async IAsyncEnumerable<MemoryRecord> GetBatchAsync(string collectionName, IEnumerable<string> keys, bool withEmbeddings = false,
         [EnumeratorCancellation] CancellationToken cancellationToken = default)
     {
-        foreach (var key in keys)
+        await foreach (var memoryRecord in this.InternalGetBatchAsync(this._dbConnection, collectionName, keys.ToArray(), withEmbeddings, cancellationToken).ConfigureAwait(false))
         {
-            var result = await this.InternalGetAsync(this._dbConnection, collectionName, key, withEmbeddings, cancellationToken).ConfigureAwait(false);
-            if (result is not null)
-            {
-                yield return result;
-            }
-            else
-            {
-                yield break;
-            }
+            yield return memoryRecord;
```

Comment:
You should be able to simplify this method further, by just returning the result of this.InternalGetBatchAsync.
You'll also need to remove the async on line 90, but then it's just a case of returning the IAsyncEnumerable directly.  No need to loop over the results and yield each.